spec/unit/spec_spec.cr

```diff
@@ -58,66 +58,59 @@ module Shards
           branch: master
         local:
           path: /var/clones/local
-          tag: unreleased
       YAML
 
       spec.dependencies.size.should eq(3)
 
       spec.dependencies[0].name.should eq("repo")
-      spec.dependencies[0].path.should be_nil
-      spec.dependencies[0].git.should eq("https://github.com/user/repo.git")
-      spec.dependencies[0].version.should eq("1.2.3")
-      spec.dependencies[0].refs.should be_nil
+      spec.dependencies[0].resolver.should eq(GitResolver.new("repo", "https://github.com/user/repo.git"))
+      spec.dependencies[0].requirement.should eq(version_req "1.2.3")
 
       spec.dependencies[1].name.should eq("example")
-      spec.dependencies[1].path.should be_nil
-      spec.dependencies[1].git.should eq("https://example.com/example-crystal.git")
-      spec.dependencies[1].version.should eq("*")
-      spec.dependencies[1].refs.should eq("master")
+      spec.dependencies[1].resolver.should eq(GitResolver.new("example", "https://example.com/example-crystal.git"))
+      spec.dependencies[1].requirement.should eq(branch "master")
 
       spec.dependencies[2].name.should eq("local")
-      spec.dependencies[2].path.should eq("/var/clones/local")
-      spec.dependencies[2].git.should be_nil
-      spec.dependencies[2].version.should eq("*")
-      spec.dependencies[2].refs.should eq("unreleased")
+      spec.dependencies[2].resolver.should eq(PathResolver.new("local", "/var/clones/local"))
+      spec.dependencies[2].requirement.should eq(Any)
     end
 
-    it "fails dependency with duplicate resolver" do
```

Comment:
Why is the "fails dependency with duplicate resolver" spec no longer needed?

Reply:
Because of the way each dependency attributes are parsed now, any other attribute will be reported as "unknown attribute" instead. After the first key/value is parsed, the rest of the parameters are delegated to the resolver.

Reply:
Please keep that sample. I suppose it would error with `Unknown attribute "gitlab"`? Which feels weird because `gitlab` is a known attribute.

Reply:
Done, it's back!